### PR TITLE
Python and Django version update in Travis and Tox

### PR DESCRIPTION
**Pull request by UsamaSadiq**
**Issue:** [BOM-1566](https://openedx.atlassian.net/browse/BOM-1566)

### Changes in Travis and Tox
- Changed `python36` with `python38`. 
- Removed `django111`, `django20`, `django21`.
- Removed unnecessary constraints and pinned `pip-tools`.
- updated version to `2.0.1`.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = py{35}-django{111}-drf{39,latest},py{35,36}-django{20,21,22}-drf{39,latest}
+envlist = py35-django22-drf{39,latest}, py38-django{22,30}
```
Comment (morenol) [inline tox.ini:2]: I think that we should remove the drf{39,  latest} part. And also the reference to drflatest in the .travis.yml file

File: .travis.yml [hunk 2]
```diff
@@ -1,27 +1,17 @@
...
-  include:
-  - python: 3.5
-    env: TOXENV=quality
-  - python: 3.5
-    env: TOXENV=docs
 cache:
 - pip
 install:
 - pip install -r requirements/travis.txt
+- pip install pip==20.0.2
```
Comment (jmbowman) [inline .travis.yml:14]: This should probably go before the line above, so dependencies are installed with the correct pip version.

File: tox.ini (hunk 1 above)
Comment (jmbowman) [reply tox.ini:2]: edx-platform is still on DRF 3.9.4, and this package is used there, so we need to keep testing for compatibility with that until it gets upgraded to a newer version.  So actually, we need to add back the `drf39` deps entry below and keep testing it in Travis.

File: .travis.yml (hunk 2 above)
Comment (UsamaSadiq) [reply .travis.yml:13]: Updated the sequence and also added back the `drf-39` in tox.

File: tox.ini [hunk 3]
```diff
@@ -18,12 +18,8 @@ setenv =
...
     DJANGO_SETTINGS_MODULE = test_settings
 deps =
-    django111: Django>=1.11,<2
-    django20: Django>=2.0,<2.1
-    django21: Django>=2.1,<2.2
     django22: Django>=2.2,<2.3
-    drf37: djangorestframework<3.8.0
-    drf38: djangorestframework<3.9.0
+    django30: Django>=3.0,<3.1
     drf39: djangorestframework<4.0.0
```
Comment (jmbowman) [inline tox.ini:23]: Sorry, one last thing I just spotted; this actually gets DRF 3.11.0, we want `djangorestframework<3.10.0` instead to get DRF 3.9.4.

Comment (UsamaSadiq) [reply tox.ini]: Updated as per suggestion.

**Pull request merged**